### ISSUE TITLE
`fft_plan_type` stores fftw_plan pointer as a fixed `int8` instead of a `c_intptr_t`

### ISSUE THREAD

**Issue by mbanck** [cp2k/cp2k#1272]
I get build failures on 32bit x86 for v8.1:

```
mpif90 -c -g -Wall -O2 -fopenmp -funroll-loops -ftree-vectorize -ffree-form -ffree-line-length-none -D__GFORTRAN -D__FFTSG -D__FFTW3 -D__FFTW3_UNALIGNED -D__LIBINT -D__LIBXC -D__SPGLIB -D__parallel -D__BLACS -D__SCALAPACK -D__ELPA=201605 -I/usr/include/elpa/modules/ -I/usr/include/elpa/elpa/ -I/usr/include -D__COMPILE_ARCH="\"Linux-i686-gfortran\"" -D__COMPILE_DATE="\"Thu Dec 31 16:51:47 UTC 2020\"" -D__COMPILE_HOST="\"kohn\"" -D__COMPILE_REVISION="\"git:0b61f2f\"" -D__DATA_DIR="\"/usr/share/cp2k\"" -D__SHORT_FILE__="\"pw/fft/fftw3_lib.F\"" -I'/<<PKGBUILDDIR>>/src/pw/fft/' -I'/<<PKGBUILDDIR>>/obj/Linux-i686-gfortran/psmp/exts/dbcsr' fftw3_lib.F90 
/<<PKGBUILDDIR>>/src/pw/fft/fftw3_lib.F:873:32:

  873 |          CALL XFFTW_EXECUTE_DFT(plan%fftw_plan, zin, xout)
      |                                1
......
 1118 |          CALL dfftw_execute_dft(fftw_plan, zin(in_offset:in_offset), zout(out_offset:out_offset))
      |                                2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
/<<PKGBUILDDIR>>/src/pw/fft/fftw3_lib.F:794:38:

  794 |                CALL XFFTW_EXECUTE_DFT(plan, input(i_off), &
      |                                      1
......
 1118 |          CALL dfftw_execute_dft(fftw_plan, zin(in_offset:in_offset), zout(out_offset:out_offset))
      |                                2      
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
/<<PKGBUILDDIR>>/src/pw/fft/fftw3_lib.F:804:35:

  804 |             CALL XFFTW_EXECUTE_DFT(plan_r, input(i_off), &
      |                                   1
......
 1118 |          CALL dfftw_execute_dft(fftw_plan, zin(in_offset:in_offset), zout(out_offset:out_offset))
      |                                2   
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
/<<PKGBUILDDIR>>/src/pw/fft/fftw3_lib.F:812:32:

  812 |          CALL XFFTW_EXECUTE_DFT(plan, input(i_off), &
      |                                1
......
 1118 |          CALL dfftw_execute_dft(fftw_plan, zin(in_offset:in_offset), zout(out_offset:out_offset))
      |                                2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
make[5]: *** [/<<PKGBUILDDIR>>/Makefile:505: fftw3_lib.o] Error 1
```

See e.g. https://buildd.debian.org/status/fetch.php?pkg=cp2k&arch=i386&ver=8.1-1&stamp=1609432216&raw=0 (but happens on 32bit ARM as well)

**Comment by mbanck** [comment]
This used to be a warning already, maybe this is due to Debian unstable now using gcc-10/gfortran-10. From https://buildd.debian.org/status/fetch.php?pkg=cp2k&arch=i386&ver=7.1-5&stamp=1605443143&raw=0:

```
Warning: Possible change of value in conversion from INTEGER(8) to INTEGER(4) at (1) [-Wconversion]
/<<PKGBUILDDIR>>/src/pw/fft/fftw3_lib.F:871:32:

  871 |          CALL XFFTW_EXECUTE_DFT(plan%fftw_plan, zin, xout)
      |                                1
......
 1116 |          CALL dfftw_execute_dft(fftw_plan, zin(in_offset:in_offset), zout(out_offset:out_offset))
      |                                2
Warning: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
```

**Comment by mbanck** [comment]
Ah, it's because I dropped `-fallow-argument-mismatch` from `FCFLAGS` due to https://github.com/cp2k/cp2k/issues/1255#issuecomment-751707266 I will put it back in, but the question is, is this a bug or expected on 32bit arches?

**Comment by dev-zero** [comment]
@mbanck I haven't looked into it, but my guess is that `plan` should be of type `int(c_intptr_t)` or `type(c_ptr)` instead of an `int(8)` to match the C side, similar to  #1242

**Comment by oschuett** [comment]
Fixed via #1705.
